test: TestShowBackup flakes on timestamp with not enough decimal places

In [this](https://teamcity.cockroachdb.com/viewLog.html?buildId=11864116&buildTypeId=Cockroach_BazelEssentialCi) test run, the timestamp `2023-09-21 01:13:25.3+00` did not have enough decimal places to match `^(.*\.[0-9]{2})[0-9]*\+00$`.

```
Failed
=== RUN   TestShowBackup
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/76aec2d78eb16f45b7d08dbac54800d8/logTestShowBackup3575638593
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/testutils/testcluster/testcluster.go:1615: ((*TestCluster).ToggleReplicateQueues)
	NOTICE: .GetStores() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().GetStores() instead.

pkg/ccl/backupccl/show_test.go:49: (TestShowBackup)
	NOTICE: .DB() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().DB() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/testutils/testcluster/testcluster.go:1615: ((*TestCluster).ToggleReplicateQueues)
	NOTICE: .GetStores() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().GetStores() instead.
    show_test.go:129: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/show_test.go:129
        	Error:      	Expected value not to be nil.
        	Test:       	TestShowBackup
        	Messages:   	2023-09-21 01:13:25.3+00 does not match ^(.*\.[0-9]{2})[0-9]*\+00$
    testutils.go:289: no Invalid Descriptors
    testutils.go:289: no Invalid Descriptors
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/76aec2d78eb16f45b7d08dbac54800d8/logTestShowBackup3575638593
--- FAIL: TestShowBackup (2.22s)
```

Jira issue: CRDB-31714

--- a/pkg/ccl/backupccl/show_test.go
+++ b/pkg/ccl/backupccl/show_test.go
@@ -120,14 +120,7 @@ ORDER BY object_type, object_name`, full)
 		{"t2", "incremental", beforeTS, incTS, "0", "false"},
 	}, res)
 
-	// Different systems output different precisions (i.e. local OS X vs CI).
-	// Truncate decimal places so Go's very rigid parsing will work.
-	// TODO(bardin): Consider using a third-party library for this, or some kind
-	// of time-freezing on the test cluster.
-	truncateBackupTimeRE := regexp.MustCompile(`^(.*\.[0-9]{2})[0-9]*\+00$`)
-	matchResult := truncateBackupTimeRE.FindStringSubmatch(beforeTS)
-	require.NotNil(t, matchResult, "%s does not match %s", beforeTS, truncateBackupTimeRE)
-	backupTime, err := time.Parse("2006-01-02 15:04:05.00", matchResult[1])
+	backupTime, err := time.Parse("2006-01-02 15:04:05.999999Z07", beforeTS)
 	require.NoError(t, err)
 	backupFolder := backupTime.Format(backupbase.DateBasedIntoFolderName)
 	resolvedBackupFolder := full + backupFolder

